rust/connlib/clients/shared/src/eventloop.rs

```diff
@@ -147,6 +147,17 @@ where
                         .send(PHOENIX_TOPIC, EgressMessages::ReuseConnection(connection));
                 }
             }
+            firezone_tunnel::ClientEvent::ResourcesChanged { resources } => {
+                // Note: This may look a bit weird: We are reading an event from the tunnel and yet delegate back to the tunnel here.
+                // Couldn't the tunnel just do this internally?
+                // Technically, yes.
+                // But, we are only accessing the callbacks here which _eventually_ will be removed from `Tunnel`.
+                // At that point, the tunnel has to emit this event and we need to handle it without delegating back to the tunnel.
+                // We only access the callbacks here because `Tunnel` already has them and the callbacks are the current way of talking to the UI.
+                // At a later point, we will probably map to another event here that gets pushed further up.
+
+                self.tunnel.callbacks.on_update_resources(resources)
```

Comment:
Is it possible to do `self.$SOMETHING.resources()` here so that the buffered event is not "Here are the new resources", but just "The resources have changed, come get the new ones"?

Reply:
We could. It would require exposing an accessor to `ClientState` or adding another forwarding getter for the resources.

I opted to just add the payload to the event for now to avoid that.

Once we have the resources (pun intended) to shuffle around some more of these abstractions, we can definitely explore what the best API is.

For example, I think it makes sense to move `PhoenixChannel` into `Tunnel`. At that point, we may able to completely remove the client `Eventloop` type because it will essentially be inside `Tunnel` already. That may allow the linux and windows client to further remove some layers at which point we may need to reconsider these APIs!